include/alpaka/vec/Vec.hpp

```diff
@@ -227,6 +227,13 @@ namespace alpaka
             return m_data[Dim::value - 4];
         }
 
+        //! Function returns the size of the static array
+        //! \return The size
+        ALPAKA_FN_HOST_ACC static constexpr decltype(auto) size()
```

Comment:
_please_ put an explicit return type.
I understand that the compiler can guess what it is, but the users shouldn't have to try and guess.

Reply:
Please copy the function signature from my PR #2415 
This is the state of the art definition for C++20.